Kudu.Core/Helpers/PostDeploymentHelper.cs

```diff
@@ -620,10 +621,37 @@ private static async Task PostAsync(string path, string requestId, string conten
                     using (var response = await client.PostAsync(path, payload))
                     {
                         statusCode = response.StatusCode;
+                        if (response.Content != null)
+                        {
+                            resContent = response.Content.ReadAsStringAsync().Result;
+                        }
+
                         response.EnsureSuccessStatusCode();
                     }
+
+                    if (path.Equals(Constants.UpdateDeployStatusPath) && resContent.Contains("Excessive SCM Site operation requests. Retry after 5 seconds"))
+                    {
+                        // Request was throttled throw an exception
+                        // If max retries aren't reached, this request will be retried
+                        Trace(TraceEventType.Information, $"Call to {path} was throttled. Setting statusCode to {HttpStatusCode.NotAcceptable}");
+
+                        statusCode = HttpStatusCode.NotAcceptable;
+                        throw new HttpRequestException();
+                    }
                 }
             }
+            catch (HttpRequestException ex)
+            {
+                if (path.Equals(Constants.UpdateDeployStatusPath, StringComparison.OrdinalIgnoreCase) && statusCode == HttpStatusCode.NotFound)
+                {
+                    // fail silently if 404 is encountered on

```

Comment:
It would be good to call out that this will only happen during a very brief period during the upgrade the very first time

Reply:
Will do.

Reply:
It could also happen if for some reason(due to a modification in FE code) the call moves down the pipeline to Kudu.